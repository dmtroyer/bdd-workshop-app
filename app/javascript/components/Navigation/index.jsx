import React, { useState } from 'react';
import {
  Collapse,
  Nav,
  NavItem,
  NavLink,
  Navbar,
  NavbarBrand,
  NavbarToggler,
} from 'reactstrap';
import { Link } from 'react-router-dom';

const Navigation = () => {
  const toggleNavbar = () => {
    setCollapsed(!collapsed);
  };
  const [collapsed, setCollapsed] = useState(true);

  return (
    <Navbar color="princess-pink" light expand="md">
      <NavbarBrand tag={Link} to="/" data-testid="landing-page-link">
        Game
      </NavbarBrand>
      <NavbarToggler onClick={toggleNavbar} className="mr-2" />
      <Collapse isOpen={!collapsed} navbar>
        <Nav navbar>
          <NavItem>
            <NavLink tag={Link} to="/about">
              About
            </NavLink>
          </NavItem>
          <NavItem>
            <NavLink tag={Link} to="/profile">
              Profile
            </NavLink>
          </NavItem>
        </Nav>
      </Collapse>
    </Navbar>
  );
};

export default Navigation;
