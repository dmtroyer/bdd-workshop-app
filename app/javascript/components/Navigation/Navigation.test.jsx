import React from 'react';
import { render, cleanup } from 'react-testing-library';
import Navigation from '.';
import { MemoryRouter } from 'react-router-dom';

describe('Navigation', () => {
  afterEach(() => {
    cleanup();
  });

  it('renders correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
