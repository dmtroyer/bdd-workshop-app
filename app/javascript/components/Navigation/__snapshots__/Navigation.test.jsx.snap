// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
<div>
  <nav
    class="navbar navbar-expand-md navbar-light bg-princess-pink"
  >
    <a
      class="navbar-brand"
      data-testid="landing-page-link"
      href="/"
    >
      Game
    </a>
    <button
      class="mr-2 navbar-toggler"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="collapse navbar-collapse"
    >
      <ul
        class="navbar-nav"
      >
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/about"
          >
            About
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/profile"
          >
            Profile
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
